
From: Geert Uytterhoeven <geert@linux-m68k.org>

Fix off-by-one error in zone size calculation (from Didier Mequignon and Petr
Stehlik)

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 /dev/null                       |    0 
 25-akpm/arch/m68k/mm/motorola.c |    2 +-
 2 files changed, 1 insertion(+), 1 deletion(-)

diff -puN arch/m68k/mm/motorola.c~m68k-mm-off-by-one arch/m68k/mm/motorola.c
--- 25/arch/m68k/mm/motorola.c~m68k-mm-off-by-one	2004-10-02 18:40:46.112915600 -0700
+++ 25-akpm/arch/m68k/mm/motorola.c	2004-10-02 18:40:46.115915144 -0700
@@ -258,7 +258,7 @@ void __init paging_init(void)
 	printk ("before free_area_init\n");
 #endif
 	zones_size[0] = (mach_max_dma_address < (unsigned long)high_memory ?
-			 mach_max_dma_address : (unsigned long)high_memory);
+			 (mach_max_dma_address+1) : (unsigned long)high_memory);
 	zones_size[1] = (unsigned long)high_memory - zones_size[0];
 
 	zones_size[0] = (zones_size[0] - PAGE_OFFSET) >> PAGE_SHIFT;
diff -L arch/m68k/mm/motorola.c.old -puN /dev/null /dev/null
_
